web/packages/design/src/Dialog/Dialog.tsx

```diff
@@ -16,42 +16,45 @@
  * along with this program.  If not, see <http://www.gnu.org/licenses/>.
  */
 
-import React from 'react';
-import PropTypes from 'prop-types';
-import styled from 'styled-components';
+import { forwardRef, PropsWithChildren } from 'react';
+import styled, { StyleFunction } from 'styled-components';
 
-import Modal from './../Modal';
+import Modal, { Props as ModalProps } from './../Modal';
 
-export default class Dialog extends React.Component {
-  render() {
-    const { children, dialogCss, className, ...modalProps } = this.props;
-    return (
-      <Modal role="dialog" {...modalProps}>
-        <ModalBox>
-          <DialogBox
-            data-testid="dialogbox"
-            dialogCss={dialogCss}
-            className={className}
-          >
-            {children}
-          </DialogBox>
-        </ModalBox>
-      </Modal>
-    );
-  }
-}
+// ModalProps enforces its own requirements with regards to the children prop through types
+// which we do not care about here, as children passed to <Dialog> are not passed directly to <Modal>.
+type ModalPropsWithoutChildren = Omit<ModalProps, 'children'>;
 
-Dialog.defaultProps = {
-  disableBackdropClick: true,
-  disableEscapeKeyDown: true,
-};
-
-Dialog.propTypes = {
-  ...Modal.propTypes,
-  children: PropTypes.node,
-  dialogCss: PropTypes.func,
-  className: PropTypes.string,
-};
+export const Dialog = forwardRef<
+  HTMLDivElement,
+  PropsWithChildren<
+    {
+      className?: string;
+      dialogCss?: StyleFunction<any>;
+    } & ModalPropsWithoutChildren
+  >
+>((props, ref) => {
+  const { children, dialogCss, className, ...modalProps } = props;
+  return (
+    <Modal disableBackdropClick disableEscapeKeyDown {...modalProps}>
```

Comment:
Why are we changing these options? How is this related?

Reply:
We had them before as: 
```tsx
Dialog.defaultProps = {
   disableBackdropClick: true,
   disableEscapeKeyDown: true,
 };
```
So the behavior stays the same.